gcloud-java-storage/src/main/java/com/google/gcloud/storage/Storage.java

```diff
@@ -508,6 +526,26 @@ public static BlobSourceOption metagenerationMatch(long metageneration) {
     public static BlobSourceOption metagenerationNotMatch(long metageneration) {
       return new BlobSourceOption(StorageRpc.Option.IF_METAGENERATION_NOT_MATCH, metageneration);
     }
+
+    static BlobSourceOption[] setGeneration(BlobId blobId, Iterable<BlobSourceOption> options) {
```

Comment:
Can you explain? I am clearly missing something, sorry...

Why can't we change `StorageImpl.copy` to pass `copyRequest.source().generation()` as the first argument to `optionMap` call?